test/NuGet.Core.Tests/NuGet.Configuration.Test/PackageSourceProviderTests.cs

```diff
@@ -2813,6 +2821,75 @@ public void LoadPackageSources_DoesNotLoadClearedSource()
             }
         }
 
+        [Fact]
+        public void LoadPackageSources_SetMaxHttpRequest()
```

Comment:
Could we have a functional test against Mock server that actually verifies the expected behavior?